activerecord/test/cases/persistence_test.rb

```diff
@@ -897,6 +897,30 @@ def test_find_via_reload
     assert_not post.new_record?
   end
 
+  def test_reload_via_querycache
+    ActiveRecord::Base.connection.enable_query_cache!
+    ActiveRecord::Base.connection.clear_query_cache
+    assert ActiveRecord::Base.connection.query_cache_enabled, 'cache should be on'
+    parrot = Parrot.create(:name => "test_querycache_parrot")
+
+    # populate the cache with the SELECT result
+    found_parrot = Parrot.find(parrot.id)
+    assert_equal parrot.id, found_parrot.id
+
+    # Manually update the 'name' attribute in the DB directly
+    assert_equal 1, ActiveRecord::Base.connection.query_cache.length
+    ActiveRecord::Base.uncached do
+      found_parrot.name = "test_querycache_parrot_wrong_name"
+      found_parrot.save
+    end
+
+    # Now reload, and verify that it gets the DB version, and not the querycache version
+    found_parrot.reload
+    assert_equal "test_querycache_parrot_wrong_name", found_parrot.name
```

Comment:
On a fussier note, can we please not expect a string containing the word "wrong". I know that'd confuse me quite entirely if I was trying to debug that assertion failing.